consensus/bor/heimdall/client.go

```diff
@@ -196,10 +239,18 @@ retryLoop:
 }
 
 // Fetch returns data from heimdall
-func Fetch[T any](ctx context.Context, client http.Client, url *url.URL) (*T, error) {
+func Fetch[T any](ctx context.Context, request *Request) (*T, error) {
+	var failed bool = true
+
+	defer func() {
+		if metrics.EnabledExpensive {
```

Comment:
Do you enable this flag? could you add log for the state of this flag somewhere at service initiation to make it more controllable?

Reply:
Yes, I've enabled it and also checked adding logs at this place. It actually prints at start saying "Enabling Metrics".